### ISSUE TITLE
Possible issue with ah_bootstrap and Python 2

### ISSUE THREAD

**Issue by saimn** [astropy/astropy-helpers#359]
From https://github.com/astropy/astropy-helpers/pull/340#issuecomment-356805492 (and following comments), it is likely that the removal of Python 2 support is the root of the issue, because of the automatic download/installation of astropy_helpers by ah_bootstrap. I think ah_bootstrap needs to be updated to install a Python 2 compatible version of astropy_helpers in this case ?

**Comment by astrofrog** [comment]
Ah we should check if astropy-helpers has ``python_requires >= 3.5`` to prevent it from getting downloaded (just checked, it doesn't so we definitely need to add).

But anyway, how is astropy-helpers 3.0rc1 getting downloaded automatically since it's just an RC?

**Comment by saimn** [comment]
Because it uses easy_install, which does not support `--pre` (or `python_requires`):
```
testtmp ❯ easy_install astropy_helpers
Searching for astropy_helpers
Reading https://pypi.python.org/simple/astropy_helpers/
Downloading https://pypi.python.org/packages/c4/0f/e577377e118c0a4c2de3d9c143c9e383053b9dc271236b53026957dae3a4/astropy-helpers-3.0rc1.tar.gz#md5=93f3f47e212ab92ea961556912ce6b6e
Best match: astropy-helpers 3.0rc1
Processing astropy-helpers-3.0rc1.tar.gz
Writing /tmp/simon/easy_install-iu9qb341/astropy-helpers-3.0rc1/setup.cfg
Running astropy-helpers-3.0rc1/setup.py -q bdist_egg --dist-dir /tmp/simon/easy_install-iu9qb341/astropy-helpers-3.0rc1/egg-dist-tmp-9yr20wuu
/tmp/simon/easy_install-iu9qb341/astropy-helpers-3.0rc1/astropy_helpers/setup_helpers.py:361: AstropyDeprecationWarning: Use of the exclude parameter is no longer supported since it does not work as expected. Use add_exclude_packages instead. Note that it must be called prior to any other calls from setup helpers.
  AstropyDeprecationWarning)
/tmp/simon/easy_install-iu9qb341/astropy-helpers-3.0rc1/astropy_helpers/setup_helpers.py:672: AstropyDeprecationWarning: Use of the exclude parameter is no longer supported since it does not work as expected. Use add_exclude_packages instead. Note that it must be called prior to any other calls from setup helpers.
  AstropyDeprecationWarning)
creating /home/simon/.pyenv/versions/3.6.3/envs/testtmp/lib/python3.6/site-packages/astropy_helpers-3.0rc1-py3.6.egg
Extracting astropy_helpers-3.0rc1-py3.6.egg to /home/simon/.pyenv/versions/3.6.3/envs/testtmp/lib/python3.6/site-packages
Adding astropy-helpers 3.0rc1 to easy-install.pth file

Installed /home/simon/.pyenv/versions/3.6.3/envs/testtmp/lib/python3.6/site-packages/astropy_helpers-3.0rc1-py3.6.egg
Processing dependencies for astropy_helpers
Finished processing dependencies for astropy_helpers
```

**Comment by astrofrog** [comment]
Ah that's unfortunate, and I guess easy_install also does not respect ``python_requires``:

```
Searching for astropy
Reading https://pypi.python.org/simple/astropy/
Downloading https://pypi.python.org/packages/29/48/f6ec337911a0abc7dbaf7a99d8e13464a4821f9e39b32bced69e510563da/astropy-3.0rc1.tar.gz#md5=648eea22ad264f7f469cbe25a5f889df
```

(astropy 3.0rc1 declares ``python_requires>3.5``)

**Comment by saimn** [comment]
Yeah. I guess we can force the version to `<3.0` if `ah_bootstrap` runs from Python 2.

**Comment by astrofrog** [comment]
In any case we should still add the ``python_requires`` to astropy-helpers before the final 3.0 release in case anyone pip installs astropy-helpers (cc @bsipocz)

**Comment by bsipocz** [comment]
I've added the 3.0.0 milestone, but suspect that we need to backport this to 2.0.x and have a helpers release there, too before 3.0

**Comment by astrofrog** [comment]
This should be fixed by #362 and #363
